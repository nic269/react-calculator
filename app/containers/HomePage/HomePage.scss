@import 'app/styles/theme';

.container {
  align-items: center;
  display: flex;
  height: 100vh;
  justify-content: center;
  width: 100vw;
  flex-direction: column;
}

.title {
  position: relative;
  text-align: center;
  text-transform: uppercase;
  margin-top: 20px;
  margin-bottom: 20px;
  font-size: 32px;
  font-weight: 400;

  &:after {
    content: "";
    height: 1px;
    width: 4rem;
    position: absolute;
    bottom: -1rem;
    left: 50%;
    transform: translateX(-50%);
    background-color: $page-title-color;
  }
}
