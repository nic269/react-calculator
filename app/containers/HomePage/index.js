/*
 * HomePage
 *
 * This is the first thing users see of our App, at the '/' route
 *
 */

import React from 'react';
import { Helmet } from 'react-helmet';

import Calculator from 'containers/Calculator/Loadable';
import styles from './HomePage.scss';

export default function HomePage() {
  return (
    <>
      <Helmet>
        <title>Calculator application</title>
        <meta name="description" content="Calculator application" />
      </Helmet>
      <div className={styles.container}>
        <Calculator />
      </div>
    </>
  );
}
