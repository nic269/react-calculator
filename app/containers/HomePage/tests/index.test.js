import React from 'react';
import { render } from 'react-testing-library';

import HomePage from '../index';

jest.mock('containers/Calculator/Loadable', () => 'Mockedcalculator');

describe('<HomePage />', () => {
  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<HomePage />);
    expect(firstChild).toMatchSnapshot();
  });
});
