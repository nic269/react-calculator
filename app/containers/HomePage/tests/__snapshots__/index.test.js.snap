// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<div>
  <h2>
    Calculator
  </h2>
  <p>
    This is a simple calculator application, you can click on the buttons of application or press keyboard also.
  </p>
  <mockedcalculator />
</div>
`;
